antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/misc/IntervalSet.kt

```diff
@@ -483,27 +463,27 @@ public open class IntervalSet : IntSet {
    * [Interval.equals] is used by the [List.equals] method to check the ranges.
    */
   override fun equals(other: Any?): Boolean =
-    other is IntervalSet && intervals == other.intervals
+    other is IntervalSet && _intervals == other._intervals
 
   override fun toString(): String =
     toString(false)
 
   public fun toString(elemAreChar: Boolean): String {
-    if (intervals == null || intervals!!.isEmpty()) {
+    if (_intervals.isEmpty()) {
       return "{}"
     }
 
-    val buf = StringBuilder()
+    val buf = StringBuilder(64)
 
     if (size() > 1) {
       buf.append("{")
     }
 
-    val iter = intervals!!.iterator()
+    val n = _intervals.size
+    var i = 0
 
-    while (iter.hasNext()) {
-      @Suppress("LocalVariableName")
-      val I = iter.next()
+    while (i < n) {
+      val I = _intervals[i++]
```

Comment:
having `i` and capital `I` is a bit confusing. I would call this one `currInterval` and `i` -> `index`

Reply:
Yeah I know, names are not the best! But I'd like to keep it as it is to match ANTLR 4.